src/main/java/graphql/schema/idl/ArgValueOfAllowedTypeChecker.java

```diff
@@ -139,66 +147,72 @@ private void checkArgValueMatchesAllowedInputType(List<GraphQLError> errors, Val
+                    .collect(joining(",")));
             return;
         }
 
         // check for unknown fields
         Map<String, InputValueDefinition> nameToInputValueDefMap = inputValueDefinitions.stream()
-                .collect(Collectors.toMap(InputValueDefinition::getName, inputValueDef -> inputValueDef));
+                .collect(toMap(InputValueDefinition::getName, inputValueDef -> inputValueDef));
 
         List<ObjectField> unknownFields = fields.stream()
                 .filter(field -> !nameToInputValueDefMap.containsKey(field.getName()))
-                .collect(Collectors.toList());
+                .collect(toList());
 
         if (!unknownFields.isEmpty()) {
             addValidationError(errors, UNKNOWN_FIELDS_MESSAGE,
                     unknownFields.stream()
                             .map(ObjectField::getName)
-                            .collect(Collectors.joining(",")),
+                            .collect(joining(",")),
                     allowedTypeDefinition.getName());
             return;
         }
 
         // fields to map for easy access
         Map<String, ObjectField> nameToFieldsMap = fields.stream()
-                .collect(Collectors.toMap(ObjectField::getName, objectField -> objectField));
+                .collect(toMap(ObjectField::getName, objectField -> objectField));
         // check each single field with its definition
         inputValueDefinitions.forEach(allowedValueDef -> {
             ObjectField objectField = nameToFieldsMap.get(allowedValueDef.getName());
             checkArgInputObjectValueFieldMatchesAllowedDefinition(errors, objectField, allowedValueDef);
         });
     }
 
-    private void checkArgValueMatchesAllowedEnum(List<GraphQLError> errors, Value instanceValue, EnumTypeDefinition allowedTypeDefinition) {
+    private void checkArgValueMatchesAllowedEnum(List<GraphQLError> errors, Value<?> instanceValue, EnumTypeDefinition allowedTypeDefinition) {
         if (!(instanceValue instanceof EnumValue)) {
             addValidationError(errors, EXPECTED_ENUM_MESSAGE, instanceValue.getClass().getSimpleName());
             return;
         }
 
         EnumValue enumValue = ((EnumValue) instanceValue);
 
-        boolean noneMatchAllowedEnumValue = allowedTypeDefinition.getEnumValueDefinitions().stream()
+        List<EnumTypeExtensionDefinition> enumExtensions = typeRegistry.enumTypeExtensions().getOrDefault(allowedTypeDefinition.getName(), emptyList());
+        Stream<EnumValueDefinition> enumExtStream = enumExtensions.stream().flatMap(enumExt -> enumExt.getEnumValueDefinitions().stream());
+        List<EnumValueDefinition> enumValueDefinitions = Stream.concat(allowedTypeDefinition.getEnumValueDefinitions().stream(), enumExtStream).collect(toList());
+
+        boolean noneMatchAllowedEnumValue = enumValueDefinitions.stream()
                 .noneMatch(enumAllowedValue -> enumAllowedValue.getName().equals(enumValue.getName()));
```

Comment:
This is the guts of the fix here.  Sorry for all the other guff (broken windows and all)

We didn't include extensions in this pre argument checking code